pkg/controller.v1/pytorch/pytorchjob_controller.go

```diff
@@ -389,8 +404,15 @@ func (r *PyTorchJobReconciler) UpdateJobStatus(job interface{},
 			}
 		} else {
 			if rtype == pytorchv1.PyTorchReplicaTypeWorker {
-				// TODO(gaocegege): Support SuccessPolicy
-				if expected == 0 {
+				worker0Completed, err := r.IsWorker0Completed(pytorchjob, replicas)
+				if err != nil {
+					logger.Warnf("check if worker 0 completed error %v", err)
+					return err
+				}
+				// Leave a succeeded condition for the following two cases:
+				// 1. If default success policy is used and worker 0 has completed.
+				// 2. If `SuccessPolicyAllWorkers` success policy is used and all workers are succeeded.
+				if expected == 0 || (worker0Completed && *pytorchjob.Spec.ElasticPolicy.SuccessPolicy != pytorchv1.SuccessPolicyAllWorkers) {
```

Comment:
What if ElasticPolicy is nil?

Reply:
in `tf-operator/pkg/apis/pytorch/v1/defaults.go`, i add code to make sure that ElasticPolicy must not be nil.

Reply:
Sorry, I cannot find the exact code. Could you point out which line in `defaults.go` set `ElasticPolicy` will not be nil?